toolkit/tools/internal/network/network_test.go

```diff
@@ -57,8 +82,8 @@ func TestDownloadFile(t *testing.T) {
 		{
 			name: "TestDownloadWithCancel",
```

Comment:
Would it make sense to add a test that calls cancel? You would probably need another file URL that includes an artificial delay in the `HandlerFunc`.

Reply:
Hmm... not a bad idea. I figured that was partly covered by the bogus cert path but I can do a dedicated one.

Reply:
Actually, will probably want to wait on some changes from https://github.com/microsoft/azurelinux/pull/9291 that @PawelWMS suggested (Specifically propagating the cancel into the http calls). Getting the timing right is going to be quite fragile until then.